app/commands/exercise/approach/link_matching_submissions.rb

```diff
@@ -0,0 +1,45 @@
+class Exercise::Approach::LinkMatchingSubmissions
+  include Mandate
+
+  initialize_with :approach
+
+  def call
+    if approach.tags.empty?
+      relink_submissions!(linked_submissions)
+    else
+      update_submissions!(linked_submissions)
+      update_submissions!(unlinked_submissions)
+    end
+  end
+
+  private
+  memoize
+  def analyzed_exercise_submissions
+    Submission.includes(:analysis).
+      where(exercise: approach.exercise).
+      where(analysis_status: :completed)
+  end
```

Comment:
Can you move this method below the next two pls?

Reply:
I'm a little wary of this:
`where(analysis_status: :completed)`

Because if we generate tags through a ML approach, it probably won't be via this analyzer.

I wonder if we should materialise tags onto a submission, and then match on tags not being nil? That probably makes SQL a bit easier etc with one less hop to the analysis record?